src/gwt/src/org/rstudio/studio/client/application/ApplicationInterrupt.java

```diff
@@ -108,6 +116,61 @@ public void onFailure()
       }
    }
    
+   public void interruptRNoDebug(final InterruptHandler handler) {
+      final int originalDebugType = errorManager_.getErrorHandlerType();
+      
+      if (originalDebugType != ErrorHandlerType.ERRORS_BREAK) {
+         interruptR(handler);
+      }
+      else {
+         errorManager_.setDebugSessionHandlerType(
```

Comment:
@jmcphers I ended up adding the RPC within `ShinyServerOperations`, suing `shiny::stopApp()` seems to work just fine!